pc2ccs/pc2v9
# I 450 add yaml load for account permissions

### Description of what the PR does

Adds support for setting account permissions using yaml.

### Issue which the PR fixes

Fixes #450

### Environment in which the PR was developed (OS,IDE, Java version, etc.)

Microsoft Windows [Version 10.0.19044.1766]

java version "1.8.0_121"
Java(TM) SE Runtime Environment (build 1.8.0_121-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.121-b13, mixed mode)

### Precise steps for _testing_ the PR (i.e., how to demonstrate that it works correctly)

This is a very general feature, it provides a way for any account or client type to either
enable or disable a permission.   Both the account number and enable/disable can 
contains lists of accounts and/or permissions respectively.

To make testing easier update/use  tenprob new permissions section (included below) to 
define which accounts are assigned particular permissions.   

 Examples of values and complete list of permission enum names are in #450 at https://github.com/pc2ccs/pc2v9/issues/450#issuecomment-1162513058

Start server using tenprob sample contest
  pc2server --login s --contestpassword contest  --load samps\contests\tenprobs

Start an Admin 1
   In edit account, edit admin 1, Verify VIEW_SECURITY_ALERTS, "View Security Alerts Log/View"  is unchecked 

To verify GIVE_RUN for judges
  Login as any judge, on the Run tab the Give button is visible 

To verify EDIT_RUN/SHADOW_PROXY_TEAM  for feeder 3
   In admin 1, In edit account, edit feeder 3, Verify EDIT_RUN, "Edit a run"  is checked
   In admin 1, In edit account, edit feeder 3, Verify SHADOW_PROXY_TEAM,"Shadow Proxy Team" is checked

To verify that for feeder 4
   In admin 1, In edit account, edit feeder 3, Verify EDIT_RUN, "Edit a run"  is not checked
   In admin 1, In edit account, edit feeder 3, Verify SHADOW_PROXY_TEAM,"Shadow Proxy Team" is not checked

--

Sample permission section from samps/contests/tenprobs/config/contest.yaml
```
# account permissions
permissions:

  - account: ADMINISTRATOR
    number: 1,2,3
    disable: VIEW_SECURITY_ALERTS

  - account: JUDGE
    number: All
    enable: GIVE_RUN

# enable EDIT_RUN for FEEDER 3
  - account: FEEDER
    number: 3
    enable: SHADOW_PROXY_TEAM, EDIT_RUN

```
